aten/src/ATen/native/cudnn/Conv.cpp

```diff
@@ -927,6 +927,8 @@ Tensor cudnn_convolution_forward(
 
   // See #4500
   Tensor weight_contig = weight->contiguous(layout);
+  // Make sure that NC11 strides follow formula
+  weight_contig.resize_(weight_contig.sizes(), layout);
```

Comment:
I understand where the problem is coming from. But I'm concerned that calling `weight->contiguous(layout)` doesn't always give you the desired memory_layout requested in `layout`.
Any reason that we are not fixing that part? to avoid the overhead in `contiguous`? I don't know much about the overhead there, I'm pulling @ngimel to get her opinion on this.

Reply:
This is to avoid putting more functionality into `contiguous` that we cannot maintain in future, and can't guarantee that strides will be correct all the time.

I tried to isolate passing information inside of 0,1-sized dims to the scope of convolution functions.